src/trajectory.jl

```diff
@@ -676,7 +677,7 @@ function mh_accept_ratio(
     Horiginal::T,
     Hproposal::T,
 ) where {T <: AbstractFloat}
-    α = min(1.0, exp(Horiginal - Hproposal))
+    α = min(one(T), exp(Horiginal - Hproposal))
     accept = rand(rng) < α
```

Comment:
@yebai I think this is a bug. We should have generate a vector of uniform R.V.s otherwise the chains are explicitly coupled. Do you agree?

Reply:
yes,  this should be a vector of IID uniform RVs. I thought `rng` is a vector of random streams, which should return a vector right?

Reply:
This one does not. The function signature is `rng::AbstractRNG`.

Reply:
Shall we fix this from this PR?